src/shared/mount-util.c

```diff
@@ -140,14 +135,9 @@ int bind_remount_recursive_with_mountinfo(
                 if (!todo)
                         return -ENOMEM;
 
-                table = mnt_new_table();
-                iter = mnt_new_iter(MNT_ITER_FORWARD);
-                if (!table || !iter)
-                        return -ENOMEM;
-
                 rewind(proc_self_mountinfo);
 
-                r = mnt_table_parse_stream(table, proc_self_mountinfo, "/proc/self/mountinfo");
+                r = libmount_parse("/proc/self/mountinfo", proc_self_mountinfo, &table, &iter);
```

Comment:
hmm, so i wonder about the two uses in bind_remount_recursive_with_mountinfo() and umount_recursive(). We use this when setting up a namespace environment for a service. i.e. we operate on a different root than the real one. libmount is unlikely to understand that though and will merge data from  /run/mount/utab into what is read here, but that's data about host mounts, not about mounts in the little namespaced environment we are setting up. Hence, we should somehow massage libmount into not loading utab meta info here if we want to use it in an entirely generic function like these two that should make no assumptions on the context they are called in.

@karelzak can you comment on this maybe? is there a way we can use the libmount mnt_table_xyz API without it reading utab for such namespaced environments?

Reply:
> @karelzak can you comment on this maybe? is there a way we can use the libmount mnt_table_xyz API without it reading utab for such namespaced environments?

Only mnt_table_parse_mtab() reads utab userspace junk.

So, all you need is to use mnt_table_parse_file(), mnt_new_table_from_file() or mnt_table_parse_stream() with "/proc/self/mountinfo" as path to the table.

And for libmount_monitor API -- the utab is enabled only if explicitly requested by  mnt_monitor_enable_userspace().